// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

syntax = "proto2";

message t_harq_stat_item {
   required uint32 count = 1;
   required uint32 total = 2;
   required uint32 min = 3;
   required uint32 max = 4;
}

message t_harq_stat_traffic_item {
   required uint32 count = 1;
   required uint32 total = 2;
}

message t_harq_type_stats {
   required uint32                   count = 1;
   required t_harq_stat_traffic_item tbs_bytes = 2;
   optional t_harq_stat_item         cqi = 3;
}

message t_harq_stats {
   required uint32             cell_id = 1;
   required uint32             rnti = 2;
   required uint32             du_ue_index = 3;

   required uint32             max_nof_harq_retxs = 4;   
   required uint32             mcs_table = 5;  

   required t_harq_stat_item   cons_retx = 6;
   required t_harq_stat_item   mcs = 7;

   repeated t_harq_type_stats  perHarqTypeStats = 8;
}

message harq_stats {
   required uint64 timestamp = 1;
   repeated t_harq_stats stats = 2;
}




